src/wallet/wallet.cpp

```diff
@@ -4475,4 +4479,197 @@ std::optional<WalletTXO> CWallet::GetTXO(const COutPoint& outpoint) const
     }
     return it->second;
 }
+
+util::Result<std::vector<WalletDescInfo>> CWallet::ExportDescriptors(bool export_private) const
```

Comment:
nit: shouldn't be something like GetDescriptorsInfo... ? or better naming... it's not doing an "export"...